### PR TITLE
Add Provider Share Subscription DTO to include ProviderEmail 

### PR DESCRIPTION
**Pull request by prchadal**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datashare/resource-manager/readme.go.md [hunk 1]
```diff
@@ -32,5 +32,5 @@ These settings apply only when `--tag=package-2019-11-01 --go` is specified on t
 Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
 
 ``` yaml $(tag) == 'package-2019-11-01' && $(go)
-output-folder: $(go-sdk-folder)/services/datashare/mgmt/2019-11-01/$(namespace)
+output-folder: $(go-sdk-folder)/services/stable/datashare/mgmt/2019-11-01/$(namespace)
```
Comment (ArcturusZhang) [inline specification/datashare/resource-manager/readme.go.md]: Please revert this change. Go package path convention does not allow extra directory between `services` and the namespace of that service, unless for preview packages.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please fix the error in go package output-folder path

**Review by ravbhatnagar — APPROVED** [review]
looks good. signing off.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
LGTM on go config files

**Review by ChenTanyi — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@prchadal please fix the CI error. Thanks.

**Review by bitsforthought — COMMENTED** [review]
Search changes are fine, since they're just whitespace.

**Comment by ChenTanyi** [comment]
@prchadal I think the "x-ms-discriminator-value" field doesn't need "required" in CI output, maybe I'll need to let ARM review again for the required field. @brjohnstmsft could you review again about these required field?

**Comment by brjohnstmsft** [comment]
@ChenTanyi I only reviewed this for the files that are part of my service (Azure Cognitive Search). I have no context on the rest of the changes.

**Comment by sarajang** [comment]
> @ChenTanyi I only reviewed this for the files that are part of my service (Azure Cognitive Search). I have no context on the rest of the changes.

@ChenTanyi  @brjohnstmsft  The required part for discriminator is coming from auto rest valdiation. the oav module before 0.20.0 didnt enforce this condition. looks like the CI for PRs got updated with the latest version of oav. Can you please review this PR with that in perspective?

File: specification/datashare/resource-manager/Microsoft.DataShare/preview/2018-11-01-preview/DataShare.json [hunk 2]
```diff
@@ -4525,7 +4544,8 @@
     "ScheduledSynchronizationSetting": {
       "description": "A type of synchronization setting based on schedule",
       "required": [
-        "properties"
+        "properties",
+        "kind"
```
Comment (ChenTanyi) [inline specification/datashare/resource-manager/Microsoft.DataShare/preview/2018-11-01-preview/DataShare.json]: There is no field name "kind" in these properties, like the next others. "x-ms-discriminator-value" is not in CI error output, just "discriminator" field is in CI error output.

**Review by ChenTanyi — COMMENTED** [review]
@sarajang yeah, I saw the "discriminator" field should be required in CI output, but the field with "x-ms-discriminator-value" doesn't show in it. Therefore, I am wondering whether this field should also be required?
@ravbhatnagar could you review again for the change of required field? Thanks!

File: specification/datashare/resource-manager/Microsoft.DataShare/preview/2018-11-01-preview/DataShare.json (hunk 2 above)
Comment (sarajang) [reply specification/datashare/resource-manager/Microsoft.DataShare/preview/2018-11-01-preview/DataShare.json]: > @sarajang yeah, I saw the "discriminator" field should be required in CI output, but the field with "x-ms-discriminator-value" doesn't show in it. Therefore, I am wondering whether this field should also be required?
> @ravbhatnagar could you review again for the change of required field? Thanks!

@ryansbenson  can you please look into this? I think this is how kind and x-ms-discriminator are handled for read only

**Review by ryansbenson — APPROVED** [review]
Approved from ARMs side

**Comment by prchadal** [comment]
@ChenTanyi Received ARM sign off, please review for the final merge off.
